server/src/services/app.js

```diff
@@ -0,0 +1,104 @@
+const {
+    createApp,
+    createAppStatus,
+    createAppVersion,
+    createLocalizedAppVersion,
+    addAppVersionToChannel,
+    addAppMedia,
+} = require('../data')
+
+exports.create = async (
+    { userId: currentUserId, organisationId, appType, status },
+    db
+) => {
```

Comment:
What's the reason for not using a transaction here? We need this to be a transaction so if something fails we have orphan tuples in the DB. Please us `db.transaction(trx =>)` to compose these. Either in the `create`-service or in the handler and pass that as the `db`-parameter to this function.